test/e2e/shared/aws.go

```diff
@@ -410,13 +410,16 @@ func createCloudFormationStack(prov client.ConfigProvider, t *cfn_bootstrap.Temp
 	cfnSvc := cloudformation.NewService(CFN)
 
 	Eventually(func() bool {
```

Comment:
This whole eventually is a wonkey. The first event that it sees is this:
```
   STEP: Loading the e2e test configuration from "/home/prow/go/src/sigs.k8s.io/cluster-api-provider-aws/test/e2e/data/e2e_eks_conf.yaml" @ 02/10/23 05:21:14.18
  STEP: Getting an AWS IAM session - from environment @ 02/10/23 05:21:14.181
  STEP: Creating a bootstrap AWSIAMConfiguration @ 02/10/23 05:21:14.182
  STEP: Creating AWS CloudFormation stack for AWS IAM resources: stack-name=cluster-api-provider-aws-sigs-k8s-io @ 02/10/23 05:21:14.188 
```

That suggests that the CF stack is already there. Yet there is no already exists error. So the CF stack must be there from a previous test run somehow and it somehow doesn't find it again or, immediately the first event it got is that. The problem is that after that it stays in RollBack. So instead of that, we Delete it and we don't roll back in a test environment. Hopefully that will help.

Reply:
BTW, create will NOT throw an error if it fails afterwards. Meaning, we also need to check what the status is and what the events are, which this eventually doesn't seem to care about. It describes events, but doesn't check if any of them is a ROLLBACK or a FAILED or anything....